http_requests.py

```diff
@@ -221,3 +226,66 @@ def delete(self, url: str, return_type=None, **kwargs):
                 return_type=None,
                 **kwargs
         )
+
+
+class EncodingMethod(enum.StrEnum):
+    GZIP = 'gzip'
+
+
+def encode_request(
+    data: str | bytes | dict | typing.IO=None,
+    json: dict=None,
+    headers: dict[str, str]=None,
+    encoding_method: EncodingMethod=EncodingMethod.GZIP,
+) -> (
+    tuple[collections.abc.Generator[bytes, None, None], dict[str, str]]
+    | collections.abc.Generator[bytes, None, None]
+):
+    '''
+    Encodes the given `data` or `json` property based on the selected `encoding_method`. If both
+    `data` and `json` are provided, the `data` property is used (same behaviour as implemented by
```

Comment:
really? this is how requests does it? that's horrifying.
I personally would prefer to raise a VE in that case to signal programming errors, tbh